src/main/java/com/fcgl/madrid/search/controller/UserSearchController.java

```diff
@@ -30,9 +27,13 @@ public void setServices(UserSearchService categoryService) {
      *
      * TODO: Based on what we find out on elastic search. We might have to make an api request to elastic search from here as well.
      */
-    @GetMapping(path = "/query")
-    public ResponseEntity<Response> addUserSearchQuery(@Valid UserSearchRequest request) {
-        //Call a function is userSearchService
-        return null;
+    @PostMapping(path = "/query")
+    public ResponseEntity<Response<String>> addUserSearchQuery(@Valid @RequestBody UserSearchRequest request) {
+        return userSearchService.addQueryToTable(request);
+    }
+
+    @GetMapping(path = "/query") //TODO returns a 500 not sure why
+    public ResponseEntity<Response<String>> getQueriesForUser(@Valid @PathVariable Long userId) {
```

Comment:
Remove @PathVariable and replace `Long userId` with UserId userid and it should remove the error

Reply:
public ResponseEntity<Response<String>> getQueriesForUser(@Valid UserId request) {

Reply:
Can you make this query sorted by date? And paginated as well? The UI team wants to have a tally of `"Recent Search"`